banzai_nres/traces.py

```diff
@@ -9,43 +9,42 @@
 import logging
 from scipy import ndimage
 import numpy as np
-from banzai_nres.fitting import fit_smooth_spline
+from numpy.polynomial.polynomial import Polynomial, polyfit
 
 logger = logging.getLogger('banzai')
 
 # Minimum separation between peaks in the image that could be separate traces
 MIN_TRACE_SEPARATION = 10
 
 # Cutoff in signal-to-noise to stop following a trace
-SIGNAL_TO_NOISE_TRACING_CUTOFF = 50
+SIGNAL_TO_NOISE_TRACING_CUTOFF = 10
 
-# Minimum half width of a feature to be considered a trace
-MIN_TRACE_HALF_WIDTH = 50
+# Maximum degree of polynomial which describes the y position of the trace as a function of pixel
+POLY_DEGREE = 5
 
 # The final trace will be +- this from the center in the y-direction
 TRACE_HALF_HEIGHT = 6
 
 
 def find_y_center(y, indices, weights=None):
     if weights is None:
-        weights = np.ones(y.shape, dtype=y.dtype)
```

Comment:
Does this break the blind solve?

Reply:
No, blind solve does not use find_y_center.

Reply:
Blind solve does use find_y_center implicitly because we use refine traces. We need to make sure this works with discretized data.

Reply:
Oh yes I see that now. Thanks. 

On that note @cmccully , it looks like TraceRefiner only uses the traces that are present on the image. Right now, calling TraceInitializer then TraceRefiner on an image is redundant (resulting in running refine_traces twice). Is TraceRefiner a place holder for something which will eventually be able to improve trace locations given a previous trace + a new image?